lib/msf/core/modules/metadata/search.rb

```diff
@@ -43,8 +43,8 @@ def self.parse_search_string(search_string)
 
     terms.each do |term|
       keyword, search_term = term.split(":", 2)
-      unless search_term
-        search_term = keyword
+      if search_term.nil? || search_term.length == 0
+        search_term = keyword.sub(/:+$/, '')
```

Comment:
Whilst this is a nice idea I'm not sure this is needed. Here were my tests. Notice that in no case did the keyword ever have a `:` within it:

```
irb(main):008:0> keyword, search_term = "asdf::".split(":", 2)
=> ["asdf", ":"]
irb(main):009:0> keyword, search_term = "asdf:::".split(":", 2)
=> ["asdf", "::"]
irb(main):010:0> keyword, search_term = "asdf:a:::".split(":", 2)
=> ["asdf", "a:::"]
irb(main):011:0> 
```

Note this is because we split on the first instance of the `:` occurring which means either there will be a string that goes into `keyword` which contains no `:` and the `search_term` will be blank, or we will split on the first instance of `:` in which case `keyword` will contain the part before the first instance of `:` and `search_term` will contain everything after it.